caddyhttp/httpserver/server.go

```diff
@@ -413,7 +413,12 @@ func (s *Server) serveHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 	// the URL path, so a request to example.com/foo/blog on the site
 	// defined as example.com/foo appears as /blog instead of /foo/blog.
 	if pathPrefix != "/" {
-		r.URL.Path = strings.TrimPrefix(r.URL.Path, pathPrefix)
+		// We need to use URL.EscapedPath() when trimming the pathPrefix as
+		// URL.Path is ambiguous about / or %2f - see docs. See #1927
+		r.URL, err = r.URL.Parse(strings.TrimPrefix(r.URL.EscapedPath(), pathPrefix))
```

Comment:
I have to reparse here as when I tried to simply assign to r.URL.Path I got unexpected results